### PR TITLE
Changed uninstall tests to be independent using a new describe().

### LINKED ISSUE #27: All tests should be independent
Tests should be able to run in arbitrary order and in isolation. The second test here seems to rely on the first test.

[source](https://github.com/marcrowo/share-button-study/blob/680e8aecd097ec65e96c79d8739bcfafc4a6ea3f/test/button_test.js#L105-L116)

```js
// ADDON UNINSTALLED
it("should no longer trigger animation once uninstalled", async() => {
  await utils.uninstallAddon(driver, addonId);
  await utils.copyUrlBar(driver);
  const { hasClass, hasColor } = await utils.testAnimation(driver);
  assert(!hasClass && !hasColor);
});

it("should no longer trigger popup once uninstalled", async() => {
  await utils.copyUrlBar(driver);
  assert(!(await utils.testPanel(driver)));
});
```

### PR DESCRIPTION
**Pull request by marcrowo**
Fixes #27.

### CODE REVIEW

File: test/button_test.js [hunk 1]
```diff
@@ -120,19 +120,28 @@ describe("Add-on Functional Tests", function() {
...
+  // This lets us assume the addon is installed at the start of each test.
+  describe("Addon uninstall tests", () => {
+    beforeEach(async() => {
+      await utils.uninstallAddon(driver, addonId);
+    });
 
-  it("should no longer trigger popup once uninstalled",
-    async() => {
+    afterEach(async() => {
+      addonId = await utils.installAddon(driver);
```
Comment (mythmon) [inline test/button_test.js]: Why are you assigning this to `addonId`?

File: test/button_test.js [hunk 2]
```diff
@@ -120,19 +120,28 @@ describe("Add-on Functional Tests", function() {
...
-  it("should no longer trigger animation once uninstalled", async() => {
-    await utils.uninstallAddon(driver, addonId);
-    await utils.copyUrlBar(driver);
-    const { hasClass, hasColor } = await utils.testAnimation(driver);
-    assert(!hasClass && !hasColor);
-  });
+  // These tests uninstall the addon before and install the addon after.
+  // This lets us assume the addon is installed at the start of each test.
+  describe("Addon uninstall tests", () => {
+    beforeEach(async() => {
```
Comment (mythmon) [inline test/button_test.js]: Does this need to be `beforeEach`? I think `before` would work just as well and be a little faster.

**Review by mythmon — CHANGES_REQUESTED** [review]

File: test/button_test.js (hunk 2 above)
Comment (marcrowo) [reply test/button_test.js]: Changing both `beforeEach` and `afterEach` to `before` and `after` will work. We need independence at the test suite level (`describe()`), not the test (`it()`) level.

File: test/button_test.js (hunk 1 above)
Comment (marcrowo) [reply test/button_test.js]: This is not necessary. I misunderstood that `addonId` is essentially a constant and does not change on a per-install basis.

**Review by mythmon — APPROVED** [review]

**Comment by marcrowo** [comment]
Rebuilt on CircleCI after I deleted the branch, which caused the test to fail. So I restored the branch to make the test be fixed (did not want commit to have x next to it).

**Pull request merged**